src/core/server/ui_settings/ui_settings_service.ts

```diff
@@ -109,15 +109,17 @@ export class UiSettingsService
 
   private validatesDefinitions() {
     for (const [key, definition] of this.uiSettingsDefaults) {
-      if (definition.schema) {
-        definition.schema.validate(definition.value, {}, `ui settings defaults [${key}]`);
+      if (!definition.schema) {
+        throw new Error(`Validation schema is not provided for [${key}] UI Setting`);
```

Comment:
another option could be to perform such check for `dev` mode only, but in general, I'd minimize divergence between `prod` & `dev`

Reply:
++ to preserve the same behavior between envs